src/detect-enip-cip-class.c

```diff
@@ -0,0 +1,102 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Philippe Antoine
+ *
+ * Set up ENIP cip class keyword
+ */
+
+#include "suricata-common.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-uint.h"
+#include "rust.h"
+
+#include "detect-enip-cip-class.h"
+
+static int g_enip_cip_class_id = 0;
+
+static void DetectEnipCipClassFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    rs_detect_u32_free(ptr);
+}
+
+/**
+ * \brief this function is used to parse enip_cip_class data into the current signature
+ *
+ * \param de_ctx pointer to the Detection Engine Context
+ * \param s pointer to the Current Signature
+ * \param rulestr pointer to the user provided enip cip_class options
+ *
+ * \retval 0 on Success
+ * \retval -1 on Failure
+ */
+static int DetectEnipCipClassSetup(DetectEngineCtx *de_ctx, Signature *s, const char *rulestr)
+{
+    if (DetectSignatureSetAppProto(s, ALPROTO_ENIP) != 0)
+        return -1;
+
+    DetectU32Data *du32 = DetectU32Parse(rulestr);
+    if (du32 == NULL) {
+        return -1;
+    }
```

Comment:
Curious: why not use SCReturnInt in these cases, too?

Reply:
I do not know what `SCReturnInt` does different... @victorjulien ?

Reply:
It's a wrapper around a `SCLogDebug` + `return` statement, so it can be helpful during debug. But we're not very consistent in using it.